codalab/lib/server_util.py

```diff
@@ -13,6 +16,37 @@
 from codalab.common import precondition
 
 
+class RateLimitExceededError(Exception):
+    pass
+
+
+def rate_limited(max_calls, per_seconds):
+    def decorate(func):
```

Comment:
I agree, one parameter would be easier to reason about. How about `max_calls_per_hour`, since I can't imagine we would want more than one message per second?